palindrome_linked_list.cpp

```diff
@@ -0,0 +1,152 @@
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+#include <iostream>
+#include <stack>
+#include <string>
+
+using namespace std;
+
+// basic linked list class
+class linkedList {
+public:
+
+	class node {
+	public:
+		char val;
+		node * next;
+		
+		node ( char c ) {
+			val = c;
+			next = NULL;
+		}
+
+	};
+	
+	node * head;
+
+	linkedList() {
+		head = NULL;
+	}
+
+	node * add( node * prev, char c ) {
+		
+		node * new_node = new node( c );
```

Comment:
Should always check for null after you allocate, unless you're using a "throwing" new operator